stub_uploader/get_version.py

```diff
@@ -72,56 +73,61 @@ def compute_incremented_version(
+    # release 5.6.7.8.YYYYMMDD.
+    specificity = len(version_base.release)
+    stub_specificity = max(specificity + 1, 4)
 
     if version_base.release < max_published.release[:specificity]:
         # Our published versions have gone too far ahead the upstream version
         # So we can't guarantee our second property.
         # In practice, this will only happen if the specificity of version_spec is
         # changed or we change our versioning scheme.
-        # For example, version_base=1.2, max_published=1.3.0.4, return 1.3.0.5
-        increment_specificity = max(increment_specificity, len(max_published.release))
-        incremented = list(max_published.release)
-        ensure_specificity(incremented, increment_specificity)
-        incremented[-1] += 1
-
-        incremented_version = Version(".".join(map(str, incremented)))
-        assert incremented_version > max_published
+        # For example, version_base=1.2, max_published=1.3.0.4, return 1.3.0.YYYYMMDD
+        stub_specificity = max(stub_specificity, len(max_published.release))
+        base_version_parts = list(max_published.release)
+
         # But can't keep versioning compatible with upstream...
-        assert incremented_version not in compatible
-        return incremented_version
+        is_compatible = False
 
-    if version_base.release > max_published.release[:specificity]:
-        # For example, version_base=1.2, max_published=1.1.0.4, return 1.2.0.0
-        incremented = list(version_base.release)
-        ensure_specificity(incremented, increment_specificity)
+    elif version_base.release > max_published.release[:specificity]:
+        # For example, version_base=1.2, max_published=1.1.0.4, return 1.2.0.YYYYMMDD
+        base_version_parts = list(version_base.release)
 
     else:
         assert version_base.release == max_published.release[:specificity]
-        # For example, version_base=1.1, max_published=1.1.0.4, return 1.1.0.5
-        incremented = list(max_published.release)
-        ensure_specificity(incremented, increment_specificity)
-        incremented[-1] += 1
+        # For example, version_base=1.1, max_published=1.1.0.YYYYMMDD (old),
+        # return 1.1.0.YYYYMMDD (today)
+        base_version_parts = list(max_published.release)
 
-    incremented_version = Version(".".join(map(str, incremented)))
-    assert incremented_version > max_published
-    assert incremented_version in compatible
-    return incremented_version
+    ensure_specificity(base_version_parts, stub_specificity)
+    new_version_parts = [*base_version_parts[:-1], int(date.strftime("%Y%m%d"))]
+    new_version = Version(".".join(map(str, new_version_parts)))
+    assert new_version > max_published
+    assert_compatibility(version_base, new_version, is_compatible)
+    return new_version
+
+
+def assert_compatibility(
+    version_base: Version, new_version: Version, is_compatible: bool
```

Comment:
nit: I like kwargs in cases where is easy to mess up order of args and types all still match